pkg/route/ingress/ingress.go

```diff
@@ -717,6 +674,64 @@ func generateRouteName(base string) string {
 	return fmt.Sprintf("%s%s", base, utilrand.String(randomLength))
 }
 
+func tlsConfigForIngress(
+	ingress *networkingv1beta1.Ingress,
+	rule *networkingv1beta1.IngressRule,
+	potentiallyNilTLSSecret *corev1.Secret,
+) (_ *routev1.TLSConfig, incomplete bool) {
+	if !needsTLSConfig(ingress, rule) {
+		return nil, false
+	}
+	// Edge: must have cert
+	// Re-Encrypt: May have cert
+	// Passthrough: Must not have cert
+	terminationPolicy := terminationPolicyForIngress(ingress)
+	if potentiallyNilTLSSecret == nil && terminationPolicy == routev1.TLSTerminationEdge {
```

Comment:
Would it be reasonable to require `spec.tls` to be specified, even if the value is `[{}]`, in order to enable TLS?

Reply:
I guess that is debatable but I would have argued that if the annotation is set, its implicit that TLS is wanted

Reply:
If it is set to a valid value ("edge", "passthrough", or "reencrypt"), it makes sense that we interpret that as enabling TLS irrespective of `spec.tls`.  If `spec.tls` is omitted and the annotation is set to an unrecognized value (I'm thinking in particular about the empty string or the string "none"), maybe it would make sense to ignore the annotation and not enable TLS.

Reply:
Okay so a valid value on the annotation or an empty element in `spec.tls` (I guess thats supposed to cover the `Allow using TLS without specifying a cert` use-case linked above)?

Reply:
Right, that makes sense to me.

Reply:
Updated